main/src/library/MutPriorityQueue.flix

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2023 Xavier deSouza
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+pub enum MutPriorityQueue[a: Type, r: Region] {
+    case MutPriorityQueue(Region[r], Ref[Array[a, r], r], Ref[Int32, r])
+}
+
+mod MutPriorityQueue {
+
+    ///
+    /// Returns a string representation of `pq`.
+    ///
+    pub def toString(pq: MutPriorityQueue[a, r]): String \ r with ToString[a] = region reg {
+        let MutPriorityQueue(_, arr, s) = pq;
+        let sb = StringBuilder.new(reg);
+        StringBuilder.appendString!("MutPriorityQueue {", sb);
+        Array.forEachWithIndex((i, x) -> {
```

Comment:
Is there not a better way? What does the other data structures do?

Reply:
I based this off of the toString functions in `MutSet`, `MutList`, `MutMap`, and `MutDeque`, which all follow a similar approach.

Reply:
Roger. We should probably simplify those as well, but I guess that can be done later.